src/enclave/enclave_signal.c

```diff
@@ -151,10 +152,11 @@ static uint64_t sgxlkl_enclave_signal_handler(
 #endif
 
         /**
-         * If LKL has not yet been initialised, we cannot handle the
-         * exception and fail instead.
+         * If LKL has not yet been initialised or is terminating (and thus no
+         * longer accepts signals), we cannot handle the exception and fail
+         * instead.
          */
-        if (!lkl_is_running())
+        if (!lkl_is_running() || is_lkl_terminating())
         {
 #ifdef DEBUG
             sgxlkl_error("Exception received before LKL can handle it. "
```

Comment:
This message is misleading now as it could either before or after LKL can handle it.

Reply:
Good point. Fixed.